qa/tasks/cephadm.py

```diff
@@ -447,6 +447,62 @@ def pull_image(ctx, config):
     finally:
         pass
 
+@contextlib.contextmanager
+def setup_ca_signed_keys(ctx, config):
+    # generate our ca key
+    cluster_name = config['cluster']
+    bootstrap_remote = ctx.ceph[cluster_name].bootstrap_remote
+    bootstrap_remote.run(args=[
+        'sudo', 'ssh-keygen', '-t', 'rsa', '-f', '/root/ca-key', '-N', '""'
+    ])
+
+    ca_key_pub_contents = bootstrap_remote.read_file('/root/ca-key.pub')
```

Comment:
```
2023-07-10T03:32:58.868 DEBUG:teuthology.orchestra.run.smithi097:> set -ex
2023-07-10T03:32:58.869 DEBUG:teuthology.orchestra.run.smithi097:> dd if=/root/ca-key.pub of=/dev/stdout
2023-07-10T03:32:58.884 DEBUG:teuthology.orchestra.run:got remote process result: 1
```
permission error I think. Need to chmod the file first if I want to use `read_file` (which seems to use `dd` without `sudo`)

Reply:
actually `/root/` where I'm writing this won't be readable by non-root anyway, even if this file individually was